### PR TITLE
Fix rootless networking with userns and ports

### PR DESCRIPTION
**Pull request by Luap99**
#### What this PR does / why we need it:
A rootless container created with a custom userns and forwarded ports
did not work. I refactored the network setup to make the setup logic
more clear.

#### How to verify it
1. terminal
`podman run -it --rm --userns keep-id --network bridge   -p 8080:8080  alpine nc -l -p 8080`
2. terminal
`echo test | nc 127.0.0.1 8080`

### CODE REVIEW

File: libpod/container_internal.go [hunk 1]
```diff
@@ -290,7 +290,7 @@ func (c *Container) handleRestartPolicy(ctx context.Context) (_ bool, retErr err
 
 	// setup slirp4netns again because slirp4netns will die when conmon exits
 	if c.config.NetMode.IsSlirp4netns() {
-		err := c.runtime.setupSlirp4netns(c)
+		err := c.runtime.setupSlirp4netns(c, c.state.NetNS)
 		if err != nil {
```
Comment (mheon) [inline libpod/container_internal.go:294]: Any reason 293 and 294 are separate lines?

**Comment by mheon** [comment]
LGTM overall

**Comment by TomSweeneyRedHat** [comment]
looks like some test unhappiness @Luap99

**Pull request merged**